Library/Homebrew/download_strategy.rb

```diff
@@ -483,7 +483,13 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
 
       if (filename_with_encoding = content_disposition.parameters["filename*"])
         encoding, encoded_filename = filename_with_encoding.split("''", 2)
-        filename = URI.decode_www_form_component(encoded_filename).encode(encoding) if encoding && encoded_filename
+        # If the `filename*` has incorrectly added double quotes, e.g.
+        #   content-disposition: attachment; filename="myapp-1.2.3.pkg"; filename*=UTF-8''"myapp-1.2.3.pkg"
+        # Then the encoded_filename will come back as the empty string, in which case we should fall back to the
+        # `filename` parameter.
+        if encoding && !encoded_filename.to_s.empty?
```

Comment:
```suggestion
        if encoding.present?
```

should be equivalent here, right? CC @Bo98 @apainintheneck

Reply:
Ahh, good to know that `.present?` exists, but it should be:

```ruby
if encoding && !encoded_filename.present?
```

no? I made that change, let me know what you think.

Reply:
That's the correct change.

Nit: You could use `encoded_filename.blank?` instead of `!encoded_filename.present?`.